app/helpers/samples_helper.rb

```diff
@@ -222,32 +207,24 @@ def filter_by_host(samples, query)
   def pipeline_run_info(pipeline_run, report_ready_pipeline_run_ids, pipeline_run_stages_by_pipeline_run_id)
     pipeline_run_entry = {}
     if pipeline_run
-      pipeline_run_entry[:job_status_description] = 'WAITING' if pipeline_run.job_status.nil?
       run_stages = pipeline_run_stages_by_pipeline_run_id[pipeline_run.id]
       if run_stages.present?
-        run_stages.each do |rs|
-          pipeline_run_entry[rs.name] = rs.job_status
-        end
         pipeline_run_entry[:total_runtime] = get_total_runtime(pipeline_run, run_stages)
         pipeline_run_entry[:with_assembly] = pipeline_run.assembly? ? 1 : 0
+        pipeline_run_entry[:result_status_description] = if pipeline_run.result_status
+                                                           pipeline_run.status_display
+                                                         else
+                                                           # data processed before result monitor was instated
+                                                           pipeline_run.status_display_pre_result_monitor(run_stages)
```

Comment:
Anything we can do to avoid maintaining the older way?   Migrating existing data perhaps?   That can happen in a separate change if needed.

Reply:
Yes. I propose after the current PR, we make an additional change where we remove the old code and use information from pipeline_run_stages to populate result_status for all the old samples.

Reply:
add a TODO so we will actually do it ;-)

Reply:
ok